docs/keycloak.md

```diff
@@ -0,0 +1,356 @@
+# Keycloak Configuration
+
+Users are registered on Keycloak, and thanks to it are authenticated and authorized
+to perform actions against the APIs provided by Orchestra Cities. The following
+section describes how Keycloak is configured and utilised to accomplish this,
+alongside the Administration Portal.
+
+## Security Configuration
+
+![Keycloak and Portal diagram](rsrc/keycloak/portal_keycloak_diagram.png)
+
+Keycloak is used to both authenticate and authorise users who wish to access the
+Orchestra Cities resources through the APIs. With the Portal for instance, we have
+two clients used: the resource-server client, and the portal client.
+The portal is deployed as a backend (providing its own API), and a frontend that users
+can access through a web browser. The frontend authenticates users via the portal
+client, and the backend authorises users via the resource-server client.
+
```

Comment:
please discuss better the security management part. 

1. i think the correct name is `resource_server`.
1. describe the role of `resource_server` as central element to govern access to entities by different services. specify that this is only available to internal apis/services to provide a security separation layer. i.e. external users cannot access it to modify attached roles / scopes.
1. introduce the other clients, e.g. api client and so on, discuss how the decoupling between them and the `resource_server` works. i.e. i ask scope `entity:read` with `api` (which has no "edit capacity") and this scope is validated against the `resource_server`, to verify is I have actually that right.

Reply:
not sure this was covered @Cerfoglg